src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1207 @@
+
+        Nonirreducible polynomials will raise an exception::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 4 - x2^2, x2: 1 + x1})
+            Traceback (most recent call last):
+            ...
+            ValueError: (LP2) fail: -x2^2 + 4 is not irreducible over Integer Ring
+
+        Different base rings are allowed::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1}, base_ring=QQ)
+            sage: S
+            A seed with cluster variables [x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+        """
+        # unpack supplied data
+
+        # make a copy
+        if isinstance(data, LPASeed):
+
+            self._names = data._names
+            self._base_ring = data._base_ring
+            self._coefficients = data._coefficients
+            self._ambient_field = data._ambient_field
+            self._polynomial_ring = data._polynomial_ring
+            self._rank = data._rank
+            self._exchange_polys = copy(data._exchange_polys)
+            self._laurent_polys = copy(data._laurent_polys)
+            self._cluster_vars = copy(data._cluster_vars)
+            self._mutation_sequence = copy(data._mutation_sequence)
+
+        # we assume we have dictionary of variable/poly pairs
+        elif isinstance(data, dict):
+
+            names = tuple(data)  # ensure names are immutable
+
+            # if we are not supplied any coefficients, check input data
+            if not coefficients:
+                for name in names:
+                    # variables present in this polynomial
+                    try:
+                        name_poly_vars = set(SR(data[name]).variables())
+                    except BaseException:
+                        raise TypeError('values of dictionary are not polynomials')
+                    # add the variables that are not already names
+                    coefficients += tuple(name_poly_vars.difference(set(names)))
+
+            coefficients = tuple(coefficients)  # coefficients are immutable
+            exchange_polys = list(data.values())
+
+            self._names = names
+
+            # construct the ambient ring from input data
+
+            # first we get generators for ambient ring
+            variables = names + coefficients
+
+            # now attempt to build polynomial ring from this data
+            if not isinstance(base_ring, (IntegerRing_class, RationalField)):
+                raise ValueError('%s is a nonsupported base ring. ZZ or QQ only.' %
+                                 (base_ring))
+
+            self._base_ring = base_ring
+            try:
+                self._polynomial_ring = PolynomialRing(self._base_ring,
+                                                       names=variables)
+            except BaseException:
+                raise TypeError('supplied variables / coefficients are not correct')
```

Comment:
```suggestion
            self._polynomial_ring = PolynomialRing(self._base_ring,
                                                       names=variables)
```
Just let it fail. The backtrace will explain where it is coming from.